frontend/src/components/teacher/session-creation/DistributeAssessmentButton.tsx

```diff
@@ -0,0 +1,58 @@
+import React from "react";
+import { useHistory } from "react-router-dom";
+import { useMutation } from "@apollo/client";
+import { Button } from "@chakra-ui/react";
+
+import { CREATE_TEST_SESSION } from "../../../APIClients/mutations/TestSessionMutations";
+import { GET_TEST_SESSIONS_BY_TEACHER_ID } from "../../../APIClients/queries/TestSessionQueries";
+import type { TestSessionRequest } from "../../../APIClients/types/TestSessionClientTypes";
+import { PaperPlaneOutlineIcon } from "../../../assets/icons";
+import { DISPLAY_ASSESSMENTS_PAGE } from "../../../constants/Routes";
+import Toast from "../../common/info/Toast";
+
+interface DistributeAssessmentButtonProps {
+  testSessionRequest: TestSessionRequest;
+}
+
+const DistributeAssessmentButton = ({
+  testSessionRequest,
+}: DistributeAssessmentButtonProps): React.ReactElement => {
+  const history = useHistory();
+  const { showToast } = Toast();
+
+  const [createSession, { loading, error }] = useMutation<{
+    createSession: string;
+  }>(CREATE_TEST_SESSION, {
+    refetchQueries: [{ query: GET_TEST_SESSIONS_BY_TEACHER_ID }],
+  });
+
+  const handleCreateSession = async (testSession: TestSessionRequest) => {
+    await createSession({ variables: { testSession } });
+    if (error) {
```

Comment:
Does this work? I feel like with the way closures work it might not as written. If you haven't yet, maybe consider throwing an error on the backend and testing that way? Usually the pattern would be to use try/catch around the `await`